cpp/src/arrow/engine/substrait/serde.h

```diff
@@ -79,150 +79,151 @@ Result<compute::ExecPlan> DeserializePlan(
 /// produced by each toplevel Substrait relation when deserializing a Substrait Plan.
 using WriteOptionsFactory = std::function<std::shared_ptr<dataset::WriteNodeOptions>()>;
 
-/// \brief Deserializes a Substrait Plan message to a list of ExecNode declarations
+/// \brief Deserializes a Substrait Plan message to a list of ExecNode declarations.
 ///
 /// The output of each top-level Substrait relation will be written to a filesystem.
 /// `write_options_factory` can be used to control write behavior.
 ///
 /// \param[in] buf a buffer containing the protobuf serialization of a Substrait Plan
-/// message
+/// message.
 /// \param[in] write_options_factory factory function for generating the write options of
-/// a node consuming the batches produced by each toplevel Substrait relation
+/// a node consuming the batches produced by each toplevel Substrait relation.
 /// \param[in] registry an extension-id-registry to use, or null for the default one.
 /// \param[out] ext_set_out if non-null, the extension mapping used by the Substrait
 /// Plan is returned here.
 /// \return a vector of ExecNode declarations, one for each toplevel relation in the
-/// Substrait Plan
+/// Substrait Plan.
 ARROW_ENGINE_EXPORT Result<std::vector<compute::Declaration>> DeserializePlans(
     const Buffer& buf, const WriteOptionsFactory& write_options_factory,
     const ExtensionIdRegistry* registry = NULLPTR, ExtensionSet* ext_set_out = NULLPTR);
 
-/// \brief Deserializes a single-relation Substrait Plan message to an execution plan
+/// \brief Deserializes a single-relation Substrait Plan message to an execution plan.
 ///
 /// The output of the single Substrait relation will be written to a filesystem.
 /// `write_options_factory` can be used to control write behavior.
 ///
 /// \param[in] buf a buffer containing the protobuf serialization of a Substrait Plan
-/// message
-/// \param[in] write_options write options of a node consuming the batches produced by
-/// each toplevel Substrait relation
+/// message.
+/// \param[in] write_options write options for the node that will consume the batches
+/// produced by the toplevel Substrait relation.
```

Comment:
```suggestion
/// produced by the top level Substrait relation.
```
Or top-level?

Reply:
That would be "top-level" (it's a compound adjective).